### PR TITLE
firestore: add empty map conformance tests

### PR DESCRIPTION
**Pull request by jadekler**
Re: https://github.com/GoogleCloudPlatform/google-cloud-python/issues/5944

### CODE REVIEW

**Comment by jadekler** [comment]
@tseaver Does this capture the case you were seeing?

**Comment by jba** [comment]
@schmidt-sebastian Please review.

File: testing/firestore/testdata/update-st-with-empty-map.textproto [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                  map_value: <
+                  >
+                >
+              >
+            >
+          >
+        >
+      >
+      update_mask: <
+        field_paths: "a.b"
```
[the remark this answers is not in the document]
Comment (jadekler) [reply testing/firestore/testdata/update-st-with-empty-map.textproto]: Good catch, thanks. Fixed.

**Comment by jadekler** [comment]
PTAL

**Comment by jadekler** [comment]
cc @jskeet @jdpedrie @blowmage

**Comment by blowmage** [comment]
These new tests passed on Ruby without needing any code changes. :+1:

**Pull request merged**